node-notes-cli/app.js

```diff
@@ -0,0 +1,150 @@
+import { writeFile } from 'node:fs/promises';
+import { readFileSync } from 'node:fs';
+
+// ------- Set up function and variables to handle user inputs -------
+const allInputs = process.argv;
+const action = process.argv[2];
+const arg3 = process.argv[3];
+const arg4 = process.argv[4];
+const legalActions = ['read', 'create', 'update', 'delete'];
+
+async function interpretInputs() {
+  try {
+    switch (action) {
+      case 'read':
+        await readNotes();
+        break;
+      case 'create':
+        await newNote(arg3);
+        break;
+      case 'update':
+        await updateNote(arg3, arg4);
+        break;
+      case 'delete':
+        await deleteNote(arg3);
+        break;
+      default:
+        console.error(action, 'is not an acceptable action. Please try again with:', legalActions);
+        process.exit(1);
+    }
+  } catch (err) {
+    handleError(err);
+  }
+}
+
+// ------- Reusable functions -------
+
+function handleError(err) {
+  console.error(err);
+  process.exit(1);
+}
+
+/** Review the ID of interest and see whether it numerical and an ID that exists within the working database, then exit the process */
```

Comment:
It's usually best to write these in the third person: "Reviews" instead of "Review". That is, you are describing what the function _does_. The one on `tooManyNotes` is good.

This is obviously a _very_ minor point.